### DISCUSSION TITLE
Widgets on resource pages

### DISCUSSION THREAD

**Discussion by codedge** [filamentphp/filament#369] [General]
I can see that widgets can be created on the dashboard, but not on resources pages. Is that something that is not supported right now or generally sth. to be thought about?

Some widgets would suit better if they would be shown at the resource page instead on the dashboard.

**Comment by countnoobula** [comment]
To my knowledge, you can add them to the header or footer. I don't think you can add them to the form layout/structure though.

**Comment by danharrin** [comment]
Yeah, there is a section in the resource docs about how to add header/footer widgets. If you want to use them in a form, use a View component and just use the `@livewire` directive to output the widget.

**Reply by countnoobula** [reply]
Thanks, hadn't thought of that ^^

**Comment by ehs4nm** [comment]
how to pass data to a widget when using from resources? 
In an edit page using getFooterWidgets() and returning the widget i need to pass the model class of current resource to the widget.
in widget class getViewData() returns data to view.

**Reply by countnoobula** [reply]
There's a small note in the docs under `Building Widgets` 
https://filamentphp.com/docs/2.x/admin/resources#building-widgets

If you define the $record field, you should have access to it in your code (on a view/edit page). I copied this from the documentation.
```
use Illuminate\Database\Eloquent\Model;
 
public ?Model $record = null;
```

**Reply by ehs4nm** [reply]
That's it. thanks.
